### ISSUE TITLE
[BUG]: excessive-permissions - false positive when permissions are set in reusable workflow parent caller

### ISSUE THREAD

**Issue by userdocs** [woodruffw/zizmor#472]
### Pre-submission checks

- [x] I am **not** filing a feature request. These should be filed via the feature request form instead.
- [x] I have looked through the [open issues](https://github.com/woodruffw/zizmor/issues?q=is%3Aissue+is%3Aopen+) for a duplicate report.

### Expected behavior

```yaml
name: reusable parent

on:
  workflow_dispatch:

jobs:
  reusable-parent:
    permissions:
      contents: write
    uses: ./.github/workflows/zizmor-child.yml
```

```yaml
name: reusable child

on:
  workflow_call:

jobs:
  reusable-child:
    runs-on: ubuntu-24.04
    steps:
      - uses: actions/checkout@v4
        with:
          persist-credentials: false

      - name: a command
        run: echo "Hello, world!"
```

Here is it in action https://github.com/userdocs/test/actions/runs/12848780414

parent caller via `workflow_dispatch` https://github.com/userdocs/test/actions/workflows/zizmor-parent.yml

### Actual behavior

```shell
[src\registry.rs:102:63] pfx = "."
warning[excessive-permissions]: overly broad permissions
  --> .github/workflows\zizmor-child.yml:7:3
   |
 7 | /   zizmor-child:
 8 | |     runs-on: ubuntu-24.04
...  |
14 | |       - name: a command
15 | |         run: echo "Hello, world!"
   | |                                  -
   | |__________________________________|
   |                                    this job
   |                                    default permissions used due to no permissions: block
   |
   = note: audit confidence → Medium
```

### Reproduction steps

Use these Workflow examples: 

https://github.com/userdocs/test/blob/main/.github/workflows/zizmor-parent.yml

https://github.com/userdocs/test/blob/main/.github/workflows/zizmor-child.yml

### Logs

```plain text

```

**Comment by woodruffw** [comment]
Thanks @userdocs!

Yeah, this is a false positive. 

This will be hard to handle in the general case (when multiple triggers are present), but in the single-trigger (`workflow_call`) case it should be sufficient to change this finding to pedantic-only. 

I'll look into a fix for that in a bit.

**Comment by woodruffw** [comment]
#473 should address this -- I've also tested it against that repo you've created and confirmed that the finding is no longer present for the regular persona.
